### ISSUE TITLE
Az cli does not refresh msgraph token cache

### ISSUE THREAD

**Issue by cveld** [Azure/azure-cli#25442]
## Describe the bug

After having consented the service principal for the msgraph app permission "Group.Read.All", the token cache is still providing a token without this permission. This results in a 403.

Even after:

```
az logout
az login -u ... -p ...
``` 

The 403 still happens.

I need to execute `az account clear` to delete the token cache, or what works better, remove the file `msal_token_cache.bin` from disk. This does not require a new login, but enforces the cli to retrieve a fresh access token.

**Command Name**
`az ad group show`

**Errors:**
```
Insufficient privileges to complete the operation.

```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- Login with a service principal without any permissions
- Add `Group.Read.All` 
- az logout
- az login
- `az ad group show --group mygroup`

Yields a 403

## Expected Behavior
The cli should flush the tokens whenever an `az logout` happens. 

## Environment Summary
```
Windows-10-10.0.22621-SP0
Python 3.10.5
Installer: MSI

azure-cli 2.40.0 *
```
## Additional Context
* `az cache list` returns an empty list interestingly
* An issue that came close to my situation; https://github.com/Azure/azure-cli/issues/17578

**Comment by yonzhan** [comment]
@jiasli for awareness

**Comment by psignoret** [comment]
I agree that it is surprising that `az logout` does not clear refresh tokens for that account. 

I also think that since `az login` obtains fresh tokens to Azure Management, it might be assumed by users that tokens to other resources (e.g. Microsoft Graph) will also be fresh tokens. This is likely what led to the confusion in https://github.com/microsoftgraph/microsoft-graph-docs/issues/22207.

**Comment by gammelgaard52** [comment]
It's still a problem though
